### PR TITLE
chore: use infura instead of public nodes

### PR DESCRIPTION
**Pull request by Sekhmet**

### CODE REVIEW

**Comment by Sekhmet** [comment]
> Can you not make it Infura specific and use STARKNET_RPC_URL instead?

Normally yes, but with it supporting multiple networks we would need to define separate env variable for each, but maybe it's what should we do - right now we need just 2.

**Comment by bonustrack** [comment]
Why would we need separate env var for each?

**Comment by Sekhmet** [comment]
> Why would we need separate env var for each?

Because the URL is different for each network (`starknet-mainnet` and `starknet-goerli`). And those subdomains are Infura specific so we can't just append that to any node URL.

**Review by bonustrack — APPROVED** [review]
Ok right forgot we need both mainnet and testnet, looks good now

File: .env.example [hunk 1]
```diff
@@ -4,6 +4,8 @@ ETH_ADDRESS=
 ETH_PRIVKEY=
 ETH_MNEMONIC=
 ETH_RPC_URL=https://rpc.brovider.xyz/5
+STARKNET_MAINNET_RPC_URL=https://starknet-mainnet.infura.io/v3/46a5dd9727bf48d4a132672d3f376146
+STARKNET_GOERLI_RPC_URL=https://starknet-goerli.infura.io/v3/46a5dd9727bf48d4a132672d3f376146
```
Comment (bonustrack) [inline .env.example:8]: Better to not expose our API key here, I've added them on DigitalOcean
```suggestion
STARKNET_MAINNET_RPC_URL=
STARKNET_GOERLI_RPC_URL=
```

Comment (bonustrack) [reply .env.example:8]: Actually nevermind, it doesn't seem to use our paid account.

Comment (Sekhmet) [reply .env.example:8]: This API key is from checkpoint-template.

**Pull request merged**